File: packages/python/src/impa/impa.py
def select_codesystem(ontology):

    if ontology == "icd9cm":
        from . import __icd9cm as icd9cm
        print("Code definitions for ICD-9-CM loaded...")
        
        return icd9cm.icd9cm

    if ontology == "icd9pcs":
        from . import __icd9pcs as icd9pcs
        print("Code definitions for ICD-9-PCS loaded...")

        return icd9pcs.icd9pcs

    if ontology == "icd10cm":
        from . import __icd10cm as icd10cm
        print("Code definitions for ICD-10-CM loaded...")

        return icd10cm.icd10cm

    if ontology == "icd10pcs":
        from . import __icd10pcs as icd10pcs
        print("Code definitions for ICD-10-PCS loaded...")

        return icd10pcs.icd10pcs

    if ontology == "icd10":
        from . import __icd10 as icd10
        print("Code definitions for ICD-10 (UK version) loaded...")

        return icd10.icd10

    if ontology == "opcs4":
        from . import __opcs4 as opcs4
        print("Code definitions for OPCS4 loaded...")

        return opcs4.opcs4

    if ontology == "cprdaurum":
        from . import __medcodeid as medcodeid
        print("Code definitions for CPRD Aurum Medcodeids loaded...")

        return medcodeid.medcodeid

File: packages/r/R/impa.R
#' @export
select_codesystem <- function(ontology){
  
  if (ontology == "icd9cm"){
    source("__icd9cm.R", local = TRUE)
    output <- generate_lookup_icd9cm()
    return(output)
  }
  
  if (ontology == "icd9pcs"){
    source("__icd9pcs.R", local = TRUE)
    output <- generate_lookup_icd9pcs()
    return(output)
  }
  
  if (ontology == "icd10cm"){
    source("__icd10cm.R", local = TRUE)
    output <- generate_lookup_icd10cm()
    return(output)
  }
  
  if (ontology == "icd10pcs"){
    source("__icd10pcs.R", local = TRUE)
    output <- generate_lookup_icd10pcs()
    return(output)
  }
  
  if (ontology == "icd10"){
    source("__icd10.R", local = TRUE)
    output <- generate_lookup_icd10()
    return(output)
  }
  
  if (ontology == "opcs4"){
    source("__opcs4.R", local = TRUE)
    output <- generate_lookup_opcs4()
    return(output)
  }
  
  if (ontology == "cprdaurum"){
    source("__medcodeid.R", local = TRUE)
    output <- generate_lookup_medcodeid()
    return(output)
  }
  
}